src/globus_sdk/paging/base.py

```diff
@@ -161,7 +161,10 @@ def decorate(func: C) -> C:
         as_paginated._paginator_items_key = items_key
         as_paginated._paginator_params = paginator_params
 
-        func.__doc__ = f"""{func.__doc__}
+        # do not append doc if the paginatedusage sphinx directive is in use
+        if ".. paginatedusage::" not in func.__doc__:
+            # but if not, append the desired text
+            func.__doc__ = f"""{func.__doc__}
```

Comment:
Although untouched by this PR, this string may need to be updated to use a `.. code-block:: pycon` as noted above.

Also, it may be useful to consider using [`.. rubric:: Paginated Usage`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-rubric) instead of simply marking the text as bold. The benefit is that it is rendered as a paragraph heading (so it _should_ be linkable) but won't generate a TOC entry.

Reply:
Both points are good and well taken. I'll switch to pycon for sure.
Regarding the rubric rule, I want to take a look at how it looks to make sure I'm okay with it, but assuming it renders similarly to the existing bold text, I'll change that too.

Reply:
I'm refraining from doing the rubric change, at least for now.
A quick experiment shows that furo styles that as uppercase text, which makes it look a little inharmonious with the surrounding (bold text) "sections" in some docs. But I've amended to switch the code block here to `pycon` styling.